<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="chromatographR">
<title>chromatographR • chromatographR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="chromatographR">
<meta property="og:description" content="chromatographR">
<meta property="og:image" content="https://ethanbass.github.io/chromatographR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">chromatographR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/chromatographR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ethanbass/chromatographR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="chromatographR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>chromatographR</h1>
                        <h4 data-toc-skip class="author">Ethan Bass</h4>
            
            <h4 data-toc-skip class="date">2022-08-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ethanbass/chromatographR/blob/HEAD/vignettes/chromatographR.Rmd" class="external-link"><code>vignettes/chromatographR.Rmd</code></a></small>
      <div class="d-none name"><code>chromatographR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em>chromatographR</em> is a package for the reproducible analysis of HPLC-DAD data in <code>R</code>. Liquid chromatography coupled to diode-array detection (HPLC-DAD) remains one of the most popular analytical methodologies due to its convenience and low-cost. However, there are currently very few open-source tools available for analyzing HPLC-DAD chromatograms or other “simple” chromatographic data. The use of proprietary software for the analysis of HPLC-DAD data is currently a significant barrier to reproducible science, since these tools are not widely accessible, and usually require users to select complicated options through a graphical user interface which cannot easily be repeated. Reproducibility is much higher in command line workflows, like <em>chromatographR</em>, where the entire analysis can be stored and easily repeated by anyone using publicly available software.</p>
<p>The <em>chromatographR</em> package draws inspiration from the previously published <em>alsace</em> package <span class="citation">(Wehrens, Carvalho, and Fraser 2015)</span>, but with improved functions for peak-finding, integration and peak table generation as well as a number of new tools for data visualization and downstream analysis. Unlike <em>alsace</em>, which emphasized multivariate curve resolution through alternating least squares (MCR-ALS), <em>chromatographR</em> is developed around a more conventional workflow that should seem more familiar to users of standard software tools for HPLC-DAD analysis. <em>chromatographR</em> includes tools for a) pre-processing, b) retention-time alignment, c) peak-finding, d) peak-integration and e) peak-table construction, as well as additional functions useful in analysis of the resulting peak table.</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>chromatographR now includes an option to directly read data from Agilent ChemStation or MassHunter files <span class="citation">(Bovee 2022a, 2022b)</span>, Chromeleon UV ASCII files, Thermo Raw, Waters ARW and others using the <code>read_chroms</code> function. Select the appropriate file format by setting the <code>format_in</code> argument to <code>csv</code>, <code>chemstation_uv</code>, <code>masshunter_dad</code>, <code>chromeleon_uv</code>, <code>waters_arw</code>, etc.</p>
<pre class="text"><code>&gt; # single folder
&gt; read_chroms(paths = path, format_in = "csv")
&gt; 
&gt; # multiple folders
&gt; path = 'foo' # path to parent directory
&gt; folders &lt;- list.files(path = path, pattern = "EXPORT3D")
&gt; dat &lt;- read_chroms(folders, format_in = "csv")</code></pre>
</div>
<div class="section level3">
<h3 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>We have included some example data consisting of root extracts from tall goldenrod (<em>Solidago altissima</em>). Roots were extracted in 90% methanol and run on an Agilent 1100 HPLC coupled to a DAD detector, according to a previously described method <span class="citation">(Uesugi and Kessler 2013)</span>. The dataset is called <code>Sa</code> (abbreviated from <em>Solidago altissima</em>).</p>
<pre class="text"><code>&gt; data(Sa)</code></pre>
</div>
<div class="section level3">
<h3 id="pre-processing-data">Pre-processing data<a class="anchor" aria-label="anchor" href="#pre-processing-data"></a>
</h3>
<p>Data from liquid chromatography often suffer from a variety of non-informative artifacts, such as noise or a drifting baseline. In addition, the data produced by the instrument may have a higher resolution or wider range (along either the time or spectral dimensions) than we require. Fortunately, most of these issues can be remedied fairly easily. For example, smoothing can reduce noise in the spectral direction while baseline subtraction can help correct a drifting baseline. Interpolation of wavelengths and retention times can be used to reduce the dimensionality of the data, facilitating comparison across samples and reducing the time and computational load required by downstream analyses. All of these functions (smoothing, baseline correction, and interpolation) are available through the <code>preprocess</code> function and are enabled by default.</p>
<p>To select a narrower range of times and/or wavelengths, arguments can be provided to <code>dim1</code> and <code>dim2</code> arguments. The <code>baseline_cor</code> function from the <em>ptw</em> package <span class="citation">(Bloemberg et al. 2010)</span> takes arguments <code>p</code> (an asymmetry parameter) and <code>lambda</code> (a smoothing parameter). You can read more about these in the documentation for <code><a href="https://rdrr.io/pkg/ptw/man/asysm.html" class="external-link">ptw::asysm</a></code>. You may want to experiment with these parameters before choosing values to use on your whole dataset.</p>
<pre class="text"><code>&gt; i=2 # chromatogram number in list of data
&gt; tpoints &lt;- as.numeric(rownames(Sa[[i]]))
&gt; lambda='200.00000'
&gt; 
&gt; matplot(x=tpoints, y=Sa[[i]][,lambda],
+        type='l', ylab='Abs (mAU)', xlab='Time (min)')
&gt; matplot(x=tpoints, y = ptw::baseline.corr(Sa[[i]][,lambda],p=.001,lambda=1e5),
+        type='l', add = T, col='blue', lty = 3)</code></pre>
<p><img src="chromatographR_files/figure-html/choosing%20parameters-1.png" width="576"></p>
<pre class="text"><code>&gt; # choose dimensions for interpolation
&gt; new.ts &lt;- seq(10,18.66,by=.01) # choose time-points
&gt; new.lambdas &lt;- seq(200, 318, by = 2) # choose wavelengths
&gt; 
&gt; dat.pr &lt;- preprocess(Sa, dim1=new.ts, dim2=new.lambdas,
+            parallel=F, p=.001, lambda=1e5)</code></pre>
</div>
<div class="section level3">
<h3 id="alignment">Alignment<a class="anchor" aria-label="anchor" href="#alignment"></a>
</h3>
<p>In many cases, liquid chromatography can suffer from retention time shifts (e.g. due to temperature fluctuations, column degradation, or subtle changes in mobile-phase composition), which can make it very difficult to compare peaks across samples. Luckily, a number of “time-warping” algorithms have been developed for correcting these kinds of shifts. In <code>chromatographR</code>, parametric time warping (<code>ptw</code>) <span class="citation">(Eilers 2004; Bloemberg et al. 2010)</span> and variable penalty dynamic time warping (<code>vpdtw</code>) <span class="citation">(Clifford et al. 2009; Clifford and Stone 2012)</span> are available for correcting retention time shifts through the <code>correct_rt</code> function. Both warping functions aim to produce a better alignment of retention times by “warping” the time-axis of each supplied chromatogram to match a reference chromatogram. (The reference chromatogram can either be determined algorithmically or selected manually by setting the <code>reference</code> argument).</p>
<div class="section level4">
<h4 id="parametric-time-warping">Parametric time warping<a class="anchor" aria-label="anchor" href="#parametric-time-warping"></a>
</h4>
<p>The <code>ptw</code> option can take a single wavelength or a list of wavelengths provided by the user using the <code>lambdas</code> argument. For each chromatogram, <code>ptw</code> then produces a “global” warping function across all the wavelengths supplied by the user. The code block below creates warping models for the samples in the provided list of data matrices. The same function is then used to warp each chromatogram according to the corresponding model, by setting the <code>models</code> parameter. Depending on the variety of your samples and the severity of the retention time shifts, it may take some experimentation with the warping parameters to get satisfactory results. Sometimes “less” can actually be “more” here – for example, wavelengths with fewer peaks may sometimes yield better warping models. (Also see the documentation for <a href="https://CRAN.R-project.org/package=ptw" class="external-link">ptw</a> for more guidance on warp function optimization).</p>
<pre class="text"><code>&gt; warping.models &lt;- correct_rt(dat.pr, what = "models", lambdas=c("210"), scale=TRUE)
&gt; warp &lt;- correct_rt(chrom_list=dat.pr, models=warping.models, what="corrected.values")</code></pre>
<p>We can then use the following code snippet to compare the alignment of warped (top panel) and unwarped (bottom panel) chromatograms.</p>
<pre class="text"><code>&gt; par(mfrow=c(2,1))
&gt; lambdas=c('210','260')
&gt; plot.new()
&gt; ts &lt;- as.numeric(rownames(warp[[i]]))
&gt; plot.window(xlim=c(head(ts,1), tail(ts,1)),ylim=c(0,1000))
&gt; for (i in 1:length(warp)){
+   matplot(ts, warp[[i]][,lambdas],type='l',add=T)
+ }
&gt; legend("topright", legend="ptw", bty = "n")
&gt; 
&gt; plot.new()
&gt; ts &lt;- as.numeric(rownames(dat.pr[[i]]))
&gt; plot.window(xlim=c(head(ts,1),tail(ts,1)),ylim=c(0,1000))
&gt; for (i in 1:length(dat.pr)){
+   matplot(ts, dat.pr[[i]][,lambdas],type='l',add=T)
+ }
&gt; legend("topright", legend="raw", bty = "n")</code></pre>
<p><img src="chromatographR_files/figure-html/alignment_plot-1.png" width="576"></p>
<p>Clearly, the alignment is considerably improved after warping. You can also use the <code>correct_rt</code> function to do a global alignment on multiple wavelengths, by providing a list of wavelengths to the <code>lambdas</code> argument, but this will not always improve the results. While the alignment still isn’t perfect after warping, it is probably good enough to align our peaks and assemble them in the peak table, which is our primary goal.</p>
</div>
<div class="section level4">
<h4 id="variable-penalty-dynamic-time-warping">Variable penalty dynamic time warping<a class="anchor" aria-label="anchor" href="#variable-penalty-dynamic-time-warping"></a>
</h4>
<p>Variable penalty dynamic time warping is another algorithm that can be very effective for correcting retention time shifts <span class="citation">(Clifford et al. 2009; Clifford and Stone 2012)</span>. Instead of calculating an explicit warping function, dynamic time warping (DTW) works by repeatedly shifting, expanding or contracting the time axis in small steps until a satisfactory alignment is attained. This process can result in very close matching of peaks, but can also lead to severe peak distortion if the time axis is repeatedly expanded and/or contracted <span class="citation">(Tomasi, van den Berg, and Andersson 2004)</span>. <strong>Variable penalty dynamic time warping</strong> is a derivative of DTW that constrains expansion and contraction of the time-axis by applying a penalty to these types of “risky” moves. This can often achieve a very good alignment with minimal peak distortion. VPdtw package is a suggested dependency since it requires a working C++ compiler. Unlike <code>ptw</code>, <code>VPdtw</code> can only take a single wavelength as input (but it returns a global warping on the whole chromatographic matrix). The code snippet below is not run, but shows how one could install <code>VPdtw</code> and use it to warp the example chromatograms.</p>
<pre class="text"><code>&gt; install.packages("VPdtw")
&gt; warp &lt;- correct_rt(chrom_list=dat.pr, alg="vpdtw", lambda="210", what="corrected.values")</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="peak-finding-and-fitting">Peak finding and fitting<a class="anchor" aria-label="anchor" href="#peak-finding-and-fitting"></a>
</h3>
<p>The <code>get_peaks</code> function produced a nested list of peaks by looping through the supplied chromatograms at the specified wavelengths, finding peaks, and fitting them to the specified function using non-linear least squares. The current peak-shape options are <code>gaussian</code> and <code>egh</code> (exponential-gaussian hybrid) <span class="citation">(Lan and Jorgenson 2001)</span>. The area under the curve for each peak is then estimated using trapezoidal approximation. The function returns a nested list of data.frames containing parameters for the peaks identified in each chromatogram.</p>
<div class="section level5">
<h5 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h5>
<p>The peak-finding algorithm may often detect a lot of peaks that are little more than noise. Thus, it is recommended to filter out extraneous peaks at this stage (especially if you are processing a lot of samples) as it can greatly reduce the computational load of peak table construction. This can be accomplished directly by using the arguments <code>sd_max</code> (to filter by peak width) and/or <code>amp_thresh</code> (to filter by peak height). Alternatively, the <code>filter_peaks</code> function can be used to filter peaks after the <code>peak_list</code> has already been created.</p>
<pre class="text"><code>&gt; pks_gauss &lt;- get_peaks(warp, lambdas = c('210','260'), sd.max=40, fit="gaussian")
&gt; pks_egh &lt;- get_peaks(warp, lambdas = c('210', '260'), sd.max=40, fit="egh")</code></pre>
</div>
<div class="section level5">
<h5 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h5>
<p>The <code>plot.peak_list</code> function allows you to visually assess the peak integration accuracy. Below we compare the peaks fitted for the first chromatogram (<code>index = 1</code>) using the two algorithms. Usually the egh function performs slightly better for asymmetrical peaks.</p>
<pre class="text"><code>&gt; par(mfrow=c(2,1))
&gt; plot(pks_gauss, index=1, lambda='210')
&gt; plot(pks_egh, index=1, lambda='210')</code></pre>
<p><img src="chromatographR_files/figure-html/plot_peaks-1.png" width="576"></p>
</div>
</div>
<div class="section level3">
<h3 id="peak-table-assembly">Peak table assembly<a class="anchor" aria-label="anchor" href="#peak-table-assembly"></a>
</h3>
<p>After obtaining a <code>peak_list</code>, the <code>get_peaks</code> function performs complete-linkage hierarchical clustering to link peaks across samples. It returns a <code>peak_table</code> object with samples as rows and peaks as columns. The <code>peak_table</code> object also has slots for holding various metadata about peaks, samples, and analyses. If you have a lot of samples, this step can be quite computationally expensive. Thus, it is suggested to filter the peak_list provided to <code>get_peaktable</code> in order to remove extraneous peaks (see <a href="###Peak%20finding%20and%20fitting">Peak finding and fitting</a> section above).</p>
<pre class="text"><code>&gt; pk_tab &lt;- get_peaktable(pks_egh, response = "area")
&gt; head(pk_tab$tab[,1:6])</code></pre>
<pre><code>          V1        V2        V3       V4        V5        V6
119 5.475999 0.0000000 0.6473611 1.889210 15.144775 0.3108266
121 4.262768 1.0941736 0.6130032 0.000000  8.006312 0.0000000
122 8.395934 0.9467302 1.9957845 0.000000 25.950200 1.7959510
458 6.201729 2.2873095 1.9889770 5.308515 13.995000 0.2875473</code></pre>
</div>
<div class="section level3">
<h3 id="further-analysis-and-data-visualization">Further analysis and data-visualization<a class="anchor" aria-label="anchor" href="#further-analysis-and-data-visualization"></a>
</h3>
<div class="section level4">
<h4 id="attaching-metadata">Attaching metadata<a class="anchor" aria-label="anchor" href="#attaching-metadata"></a>
</h4>
<p>To begin analyzing your peak table, you will usually want to attach sample meta-data to the <code>peak_table</code> object. This can be easily accomplished using the <code>attach_metadata</code> function. The peak table can then be normalized (e.g. by dividing out the sample weight) using the <code>normalize_data</code> function.</p>
<pre class="text"><code>&gt; path &lt;- system.file("extdata", "Sa_metadata.csv", package = "chromatographR")
&gt; meta &lt;- read.csv(path)
&gt; pk_tab &lt;- attach_metadata(peak_table = pk_tab, metadata = meta, column="vial")
&gt; pk_tab &lt;- normalize_data(peak_table = pk_tab, column="mass")</code></pre>
</div>
<div class="section level4">
<h4 id="data-exploration">Data exploration<a class="anchor" aria-label="anchor" href="#data-exploration"></a>
</h4>
<div class="section level5">
<h5 id="mirror-plot">Mirror plot<a class="anchor" aria-label="anchor" href="#mirror-plot"></a>
</h5>
<p>The <code>mirror_plot</code> function provides a quick way to visually compare results across treatment groups.</p>
<pre class="text"><code>&gt; mirror_plot(pk_tab, lambdas = c("210","260"), var = "trt", legend_size=2)</code></pre>
<p><img src="chromatographR_files/figure-html/unnamed-chunk-4-1.png" width="576"></p>
</div>
<div class="section level5">
<h5 id="plotting-spectra">Plotting spectra<a class="anchor" aria-label="anchor" href="#plotting-spectra"></a>
</h5>
<p>The <code>plot_spectrum</code> function allows you to easily plot or record the spectra associated with a particular peak in your peak table. This is useful for interpreting your results and/or checking for errors in your peak table. For example, you may want to check if the spectra for a particular peak match across different samples, or you may want to compare your spectrum with a known standard. The <code>plot_spectrum</code> function can be used to plot only the spectrum or only the chromatographic trace using the arguments <code>plot_spectrum</code> and <code>plot_trace</code>. By default it will plot the trace and spectrum from the chromatogram with the largest peak in the peak table. Alternatively, you can choose the chromatogram index and wavelength using the <code>chr</code> and <code>lambda</code> arguments.</p>
<pre class="text"><code>&gt; oldpar &lt;- par(no.readonly = TRUE)
&gt; par(mfrow=c(2,1))
&gt; peak="V7"
&gt; plot_spectrum(peak, peak_table = pk_tab, chrom_list=warp, verbose=F)</code></pre>
<p><img src="chromatographR_files/figure-html/plot%20spectra-1.png" width="576"></p>
<pre class="text"><code>&gt; par(oldpar)</code></pre>
<p>The <code>plot_all_spectra</code> function can be used to visually compare the spectra for a specified peak across all samples.</p>
<pre class="text"><code>&gt; peak="V13"
&gt; plot_all_spectra(peak, peak_table=pk_tab, export=F, overlapping=T)</code></pre>
<p><img src="chromatographR_files/figure-html/plot_all-1.png" width="576"></p>
</div>
<div class="section level5">
<h5 id="plot-peak-table-function">Plot peak table function<a class="anchor" aria-label="anchor" href="#plot-peak-table-function"></a>
</h5>
<p>The <code>plot.peak_table</code> function provides a simplified interface to various options for plotting data from the <code>peak_table</code>. For example, it can be used as a quick interface to the <code>plot_spectrum</code> and <code>plot_all_spectra</code> functions shown above. It can also be used to quickly compare results across treatments by calling <code>boxplot</code>.</p>
<pre class="text"><code>&gt; plot(pk_tab, loc = "V13", box_plot = TRUE, vars = "trt", verbose = FALSE)</code></pre>
<p><img src="chromatographR_files/figure-html/unnamed-chunk-5-1.png" width="576"><img src="chromatographR_files/figure-html/unnamed-chunk-5-2.png" width="576"><img src="chromatographR_files/figure-html/unnamed-chunk-5-3.png" width="576"></p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs">
<div id="ref-bloemberg2010">
<p>Bloemberg, Tom G., Jan Gerretzen, Hans J. P. Wouters, Jolein Gloerich, Maurice van Dael, Hans J. C. T. Wessels, Lambert P. van den Heuvel, Paul H. C. Eilers, Lutgarde M. C. Buydens, and Ron Wehrens. 2010. “Improved Parametric Time Warping for Proteomics.” <em>Chemometrics and Intelligent Laboratory Systems</em>, OMICS, 104 (1): 65–74. <a href="https://doi.org/10.1016/j.chemolab.2010.04.008" class="external-link">https://doi.org/10.1016/j.chemolab.2010.04.008</a>.</p>
</div>
<div id="ref-bovee2022">
<p>Bovee, Roderick. 2022a. “Aston.”</p>
</div>
<div id="ref-bovee2022a">
<p>———. 2022b. “Entab.”</p>
</div>
<div id="ref-clifford2012">
<p>Clifford, David, and Glenn Stone. 2012. “Variable Penalty Dynamic Time Warping Code for Aligning Mass Spectrometry Chromatograms in R.” <em>Journal of Statistical Software</em> 47 (April): 1–17. <a href="https://doi.org/10.18637/jss.v047.i08" class="external-link">https://doi.org/10.18637/jss.v047.i08</a>.</p>
</div>
<div id="ref-clifford2009">
<p>Clifford, David, Glenn Stone, Ivan Montoliu, Serge Rezzi, François-Pierre Martin, Philippe Guy, Stephen Bruce, and Sunil Kochhar. 2009. “Alignment Using Variable Penalty Dynamic Time Warping.” <em>Analytical Chemistry</em> 81 (3): 1000–1007. <a href="https://doi.org/10.1021/ac802041e" class="external-link">https://doi.org/10.1021/ac802041e</a>.</p>
</div>
<div id="ref-eilers2004">
<p>Eilers, Paul H. C. 2004. “Parametric Time Warping.” <em>Analytical Chemistry</em> 76 (2): 404–11. <a href="https://doi.org/10.1021/ac034800e" class="external-link">https://doi.org/10.1021/ac034800e</a>.</p>
</div>
<div id="ref-lan2001">
<p>Lan, Kevin, and James W. Jorgenson. 2001. “A Hybrid of Exponential and Gaussian Functions as a Simple Model of Asymmetric Chromatographic Peaks.” <em>Journal of Chromatography A</em> 915 (1): 1–13. <a href="https://doi.org/10.1016/S0021-9673(01)00594-5" class="external-link">https://doi.org/10.1016/S0021-9673(01)00594-5</a>.</p>
</div>
<div id="ref-tomasi2004">
<p>Tomasi, Giorgio, Frans van den Berg, and Claus Andersson. 2004. “Correlation Optimized Warping and Dynamic Time Warping as Preprocessing Methods for Chromatographic Data.” <em>Journal of Chemometrics</em> 18 (5): 231–41. <a href="https://doi.org/10.1002/cem.859" class="external-link">https://doi.org/10.1002/cem.859</a>.</p>
</div>
<div id="ref-uesugi2013">
<p>Uesugi, Akane, and André Kessler. 2013. “Herbivore Exclusion Drives the Evolution of Plant Competitiveness via Increased Allelopathy.” <em>New Phytologist</em> 198 (3): 916–24. <a href="https://doi.org/10.1111/nph.12172" class="external-link">https://doi.org/10.1111/nph.12172</a>.</p>
</div>
<div id="ref-wehrens2015">
<p>Wehrens, Ron, Elisabete Carvalho, and Paul D. Fraser. 2015. “Metabolite Profiling in LCDAD Using Multivariate Curve Resolution: The Alsace Package for R.” <em>Metabolomics</em> 11 (1): 143–54. <a href="https://doi.org/10.1007/s11306-014-0683-5" class="external-link">https://doi.org/10.1007/s11306-014-0683-5</a>.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre class="text"><code>&gt; sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] chromatographR_0.4.4 knitr_1.40          

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9            lattice_0.20-45       RcppDE_0.1.6         
 [4] tidyr_1.2.0           png_0.1-7             zoo_1.8-10           
 [7] rprojroot_2.0.3       digest_0.6.29         utf8_1.2.2           
[10] R6_2.5.1              cellranger_1.1.0      VPdtw_2.1-14         
[13] dynamicTreeCut_1.63-1 evaluate_0.16         highr_0.9            
[16] pillar_1.8.1          rlang_1.0.5           curl_4.3.2           
[19] readxl_1.4.1          TTR_0.24.3            jquerylib_0.1.4      
[22] Matrix_1.4-1          reticulate_1.25       rmarkdown_2.16       
[25] pkgdown_2.0.6         textshaping_0.3.6     desc_1.4.1           
[28] stringr_1.4.1         munsell_0.5.0         compiler_4.2.1       
[31] xfun_0.32             pkgconfig_2.0.3       systemfonts_1.0.4    
[34] htmltools_0.5.3       tidyselect_1.1.2      tibble_3.1.8         
[37] fansi_1.0.3           dplyr_1.0.9           grid_4.2.1           
[40] jsonlite_1.8.0        lifecycle_1.0.1       magrittr_2.0.3       
[43] scales_1.2.1          cli_3.3.0             stringi_1.7.8        
[46] cachem_1.0.6          farver_2.1.1          ptw_1.9-16           
[49] fs_1.5.2              xml2_1.3.3            smoother_1.1         
[52] bslib_0.4.0           fastcluster_1.2.3     xts_0.12.1           
[55] ragg_1.2.2            generics_0.1.3        vctrs_0.4.1          
[58] tools_4.2.1           pvclust_2.2-0         glue_1.6.2           
[61] purrr_0.3.4           fastmap_1.1.0         yaml_2.3.5           
[64] colorspace_2.0-3      memoise_2.0.1         minpack.lm_1.2-2     
[67] chromConverter_0.2.1  sass_0.4.2           </code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ethan Bass.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
