<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>chromatographR: An introduction to HPLC-DAD analysis • chromatographR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="chromatographR: An introduction to HPLC-DAD analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">chromatographR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.7.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/chromatographR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/FAQ.html">Frequently asked questions</a></li>
    <li><a class="dropdown-item" href="../articles/GC-FID.html">Analysis of *Polistes* cuticular hydrocarbons with GC-FID</a></li>
    <li><a class="dropdown-item" href="../articles/uv_spectra.html">Programmatic analysis of UV spectra</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ethanbass/chromatographR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>chromatographR: An introduction to HPLC-DAD analysis</h1>
                        <h4 data-toc-skip class="author">Ethan
Bass<sup>1</sup>
</h4>
            
            <h4 data-toc-skip class="date">2024-11-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ethanbass/chromatographR/blob/master/vignettes/chromatographR.Rmd" class="external-link"><code>vignettes/chromatographR.Rmd</code></a></small>
      <div class="d-none name"><code>chromatographR.Rmd</code></div>
    </div>

    
    
<p><sup>1</sup> Department of Ecology and Evolutionary Biology, Cornell
University, Ithaca NY</p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em>chromatographR</em> is a package for the reproducible analysis of
HPLC-DAD data in <code>R</code>. Liquid chromatography coupled to
diode-array detection (HPLC-DAD) remains one of the most popular
analytical methodologies due to its convenience and low-cost. However,
there are currently very few open-source tools available for analyzing
HPLC-DAD chromatograms or other “simple” chromatographic data. The use
of proprietary software for the analysis of HPLC-DAD data is currently a
significant barrier to reproducible science, since these tools are not
widely accessible, and usually require users to select complicated
options through a graphical user interface which cannot easily be
repeated. Reproducibility is much higher in command line workflows, like
<em>chromatographR</em>, where the entire analysis can be stored and
easily repeated by anyone using publicly available software.</p>
<p>The <em>chromatographR</em> package began as a fork from the
previously published <em>alsace</em> package <span class="citation">(Wehrens, Carvalho, and Fraser 2015)</span>, but has
been reworked with improved functions for peak-finding, integration and
peak table generation as well as a number of new tools for data
visualization and downstream analysis. Unlike <em>alsace</em>, which
emphasized multivariate curve resolution through alternating least
squares (MCR-ALS), <em>chromatographR</em> is developed around a more
conventional workflow that should seem more familiar to users of
standard software tools for HPLC-DAD analysis. <em>chromatographR</em>
includes tools for a) pre-processing, b) retention-time alignment, c)
peak-finding, d) peak-integration and e) peak-table construction, as
well as additional functions useful for analyzing the resulting peak
table.</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>chromatographR can import data from a growing list of proprietary
file formats using the <code>read_chroms</code> function. Supported file
formats include ‘Agilent ChemStation’ and ‘MassHunter’ (<code>.D</code>)
files, ‘Thermo Raw’ (<code>.raw</code>), ‘Chromeleon’ UV ASCII
(<code>.txt</code>), ‘Waters ARW’ (<code>.arw</code>), ‘Shimadzu’ ASCII
(<code>.txt</code>), and more. (For a full list, see the <a href="https://ethanbass.github.io/chromConverter/" class="external-link">chromConverter
documentation</a>). Select the appropriate file format by specifying the
<code>format_in</code> argument (e.g. <code>csv</code>,
<code>chemstation_uv</code>, <code>masshunter_dad</code>,
<code>chromeleon_uv</code>, <code>waters_arw</code>, etc).</p>
<pre class="text"><code>&gt; # single folder
&gt; read_chroms(paths = path, format_in = "chemstation_uv")
&gt; 
&gt; # multiple folders
&gt; path &lt;- 'foo' # path to parent directory
&gt; folders &lt;- list.files(path = path, full.names = TRUE)
&gt; dat &lt;- read_chroms(folders, format_in = "chemstation_uv")</code></pre>
</div>
<div class="section level3">
<h3 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>The package includes some example data consisting of root extracts
from tall goldenrod (<em>Solidago altissima</em>). Roots were extracted
in 90% methanol and run on an Agilent 1100 HPLC coupled to a DAD
detector, according to a previously described method <span class="citation">(Uesugi and Kessler 2013)</span>. The dataset is called
<code>Sa</code> (abbreviated from <em>Solidago altissima</em>).</p>
<pre class="text"><code>&gt; data(Sa)</code></pre>
</div>
<div class="section level3">
<h3 id="pre-processing-data">Pre-processing data<a class="anchor" aria-label="anchor" href="#pre-processing-data"></a>
</h3>
<p>Data from liquid chromatography often suffer from a variety of
non-informative artifacts, such as noise or a drifting baseline. In
addition, the data produced by the instrument may have a higher
resolution or wider range (along either the time or spectral dimensions)
than we require. Fortunately, most of these issues can be remedied
fairly easily. For example, smoothing can reduce noise in the spectral
direction while baseline subtraction can help correct a drifting
baseline. Interpolation of wavelengths and retention times can be used
to reduce the dimensionality of the data, facilitating comparison across
samples and reducing the time and computational load required by
downstream analyses. All of these functions (smoothing, baseline
correction, and interpolation) are available through the
<code>preprocess</code> function and are enabled by default.</p>
<p>To select a narrower range of times and/or wavelengths, arguments can
be provided to the optional <code>dim1</code> and <code>dim2</code>
arguments. The <code>baseline_cor</code> function from the <em>ptw</em>
package <span class="citation">(Bloemberg et al. 2010)</span> takes
arguments <code>p</code> (an asymmetry parameter) and
<code>lambda</code> (a smoothing parameter). You can read more about
these in the documentation for <code><a href="https://rdrr.io/pkg/ptw/man/asysm.html" class="external-link">ptw::asysm</a></code>. You may want to
experiment with these parameters before choosing values to use on your
whole dataset.</p>
<pre class="text"><code>&gt; i &lt;- 2 # chromatogram number in list of data
&gt; tpoints &lt;- as.numeric(rownames(Sa[[i]]))
&gt; lambda &lt;- '200.00000'
&gt; 
&gt; matplot(x = tpoints, y = Sa[[i]][,lambda],
+        type = 'l', ylab = 'Abs (mAU)', xlab = 'Time (min)')
&gt; matplot(x = tpoints, y = ptw::baseline.corr(Sa[[i]][,lambda], p = .001, lambda = 1e5),
+        type = 'l', add = TRUE, col='blue', lty = 3)</code></pre>
<p><img src="chromatographR_files/figure-html/choose_parameters_for_fitting_baseline-1.png" width="600px"></p>
<p>After selecting parameters for baseline correction, you can proceed
with the pre-processing step as shown below.</p>
<pre class="text"><code>&gt; # choose dimensions for interpolation
&gt; new.ts &lt;- seq(10, 18.66, by = .01) # choose time-points
&gt; new.lambdas &lt;- seq(200, 318, by = 2) # choose wavelengths
&gt; 
&gt; dat.pr &lt;- preprocess(Sa, dim1 = new.ts, dim2 = new.lambdas, p = .001, lambda = 1e5, cl = 1)</code></pre>
<pre><code>Warning in preprocess(Sa, dim1 = new.ts, dim2 = new.lambdas, p = 0.001, : The `parallel` argument is deprecated. Just use the `cl` argument to enable
            parallel processing.</code></pre>
</div>
<div class="section level3">
<h3 id="alignment">Alignment<a class="anchor" aria-label="anchor" href="#alignment"></a>
</h3>
<p>In many cases, liquid chromatography can suffer from retention time
shifts (e.g. due to temperature fluctuations, column degradation, or
subtle changes in mobile-phase composition), which can make it very
difficult to compare peaks across samples. Luckily, a number of
“time-warping” algorithms have been developed for correcting these kinds
of shifts. In <code>chromatographR</code>, parametric time warping
(<code>ptw</code>) <span class="citation">(Eilers 2004; Bloemberg et al.
2010)</span> and variable penalty dynamic time warping
(<code>vpdtw</code>) <span class="citation">(Clifford et al. 2009;
Clifford and Stone 2012)</span> are available for correcting retention
time shifts through the <code>correct_rt</code> function. Both warping
functions aim to produce a better alignment of features by “warping” the
time-axis of each supplied chromatogram to match a reference
chromatogram. (The reference chromatogram can either be determined
algorithmically or selected manually by setting the
<code>reference</code> argument).</p>
<p>First, we check the alignment of our chromatograms using the
<code>plot_chroms</code> function.</p>
<pre class="text"><code>&gt; plot_chroms(dat.pr, lambdas = 210)</code></pre>
<p><img src="chromatographR_files/figure-html/check%20alignment-1.png" width="600px"></p>
<p>Our chromatograms appear to be shifted, as shown by the poor overlap
of the chromatograms across our four samples. To remedy this problem, we
can try to “warp” our chromatograms using one of the two options
described above.</p>
<div class="section level4">
<h4 id="parametric-time-warping">Parametric time warping<a class="anchor" aria-label="anchor" href="#parametric-time-warping"></a>
</h4>
<p>The <code>ptw</code> option can take a single wavelength or a list of
wavelengths provided by the user using the <code>lambdas</code>
argument. For each chromatogram, <code>ptw</code> then produces a
“global” warping function across all the wavelengths supplied by the
user. The code block below creates warping models for the samples in the
provided list of data matrices. The same function is then used to warp
each chromatogram according to the corresponding model, by setting the
<code>models</code> parameter. Depending on the variety of your samples
and the severity of the retention time shifts, it may take some
experimentation with the warping parameters to get satisfactory results.
Sometimes less can actually be more here – for example, wavelengths with
fewer peaks may sometimes yield better warping models. (Also see the
documentation for <a href="https://CRAN.R-project.org/package=ptw" class="external-link">ptw</a> for more guidance
on warp function optimization).</p>
<pre class="text"><code>&gt; warping.models &lt;- correct_rt(dat.pr, what = "models", lambdas = c(210), scale = TRUE)
&gt; warp &lt;- correct_rt(chrom_list = dat.pr, models = warping.models, what = "corrected.values")</code></pre>
<p>We can then use the following code snippet to compare the alignment
of warped (top panel) and unwarped (bottom panel) chromatograms.</p>
<pre class="text"><code>&gt; par(mfrow=c(2,1))
&gt; # plot warped data
&gt; plot_chroms(warp, lambdas = c(210))
&gt; legend("topleft", legend = "ptw", bty = "n")
&gt; 
&gt; # plot unwarped data
&gt; plot_chroms(dat.pr, lambdas = c(210))
&gt; legend("topleft", legend = "raw", bty = "n")</code></pre>
<p><img src="chromatographR_files/figure-html/alignment_plot-1.png" width="600px"></p>
<p>Clearly, the alignment is considerably improved after warping. You
can also use the <code>correct_rt</code> function to do a global
alignment on multiple wavelengths, by providing a list of wavelengths to
the <code>lambdas</code> argument, but this will not always improve the
results. While the alignment still isn’t perfect after warping, it is
probably good enough to align our peaks and assemble them in the peak
table, which is our primary goal.</p>
</div>
<div class="section level4">
<h4 id="variable-penalty-dynamic-time-warping">Variable penalty dynamic time warping<a class="anchor" aria-label="anchor" href="#variable-penalty-dynamic-time-warping"></a>
</h4>
<p>Variable penalty dynamic time warping is another algorithm that can
be very effective for correcting retention time shifts <span class="citation">(Clifford et al. 2009; Clifford and Stone 2012)</span>.
Instead of calculating an explicit warping function, dynamic time
warping (DTW) works by repeatedly shifting, expanding or contracting the
time axis in small steps until a satisfactory alignment is attained.
This process can result in very close matching of peaks, but can also
lead to severe peak distortion if the time axis is repeatedly expanded
and/or contracted <span class="citation">(Tomasi, van den Berg, and
Andersson 2004)</span>. <strong>Variable penalty dynamic time
warping</strong> is a derivative of DTW that constrains expansion and
contraction of the time-axis by applying a penalty to these types of
“risky” moves. This can often achieve a very good alignment with minimal
peak distortion. VPdtw package is a suggested dependency since it
requires a working C++ compiler. Unlike <code>ptw</code>,
<code>VPdtw</code> can only take a single wavelength as input (but it
returns a global warping on the whole chromatographic matrix). The code
snippet below uses VPdtw to warp the example chromatograms using the
first chromatogram as a reference. The <code>penalty</code> and
<code>maxshift</code> parameters can be adjusted to control the degree
of warping.</p>
<pre class="text"><code>&gt; warp &lt;- correct_rt(chrom_list = dat.pr, alg = "vpdtw", lambdas = 210, 
+                    what = "corrected.values", reference = 1)</code></pre>
<pre class="text"><code>&gt; par(mfrow=c(2,1))
&gt; plot_chroms(warp, lambdas = c(210))
&gt; legend("topleft", legend = "vpdtw", bty = "n")
&gt; 
&gt; plot_chroms(dat.pr, lambdas = c(210))
&gt; legend("topleft", legend = "raw", bty = "n")</code></pre>
<p><img src="chromatographR_files/figure-html/plot_chroms-1.png" width="600px"></p>
<p>In this case the alignments produced by the two algorithms are very
similar, but in some cases one algorithm may dramatically outperform the
other.</p>
</div>
</div>
<div class="section level3">
<h3 id="peak-detection-and-integration">Peak detection and integration<a class="anchor" aria-label="anchor" href="#peak-detection-and-integration"></a>
</h3>
<p>The <code>get_peaks</code> function produced a nested list of peaks
by looping through the supplied chromatograms at the specified
wavelengths, finding peaks, and fitting them to the specified function
using non-linear least squares. The area under the curve for each peak
is then estimated using trapezoidal approximation. The <code>fit</code>
argument can be used to specify a peak-fitting model. The current
options are exponential-gaussian hybrid (<code>egh</code>) <span class="citation">(Lan and Jorgenson 2001)</span> (the default setting)
or <code>gaussian</code>. Alternatively, peak areas can be integrated
without applying a model (<code>fit = raw</code>). The function returns
a nested list of data.frames containing parameters for the peaks
identified in each chromatogram.</p>
<pre class="text"><code>&gt; # find and integrate peaks using gaussian peak fitting
&gt; pks_gauss &lt;- get_peaks(warp, lambdas = c(210), sd.max = 40, fit = "gaussian")
&gt; 
&gt; # find and integrate peaks using exponential-gaussian hybrid model
&gt; pks_egh &lt;- get_peaks(warp, lambdas = c(210), sd.max = 40, fit = "egh")
&gt; 
&gt; # find and integrate peaks without modeling peak shape
&gt; pks_raw &lt;- get_peaks(warp, lambdas = c(210), sd.max = 100, fit = "raw")</code></pre>
<div class="section level4">
<h4 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h4>
<p>The peak-finding algorithm may often detect a lot of peaks that are
little more than noise. Thus, it is recommended to filter out extraneous
peaks at this stage (especially if you are processing a lot of samples)
as it can greatly reduce the computational load of peak table
construction. This can be accomplished directly by using the arguments
<code>sd_max</code> (to filter by peak width) and/or
<code>amp_thresh</code> (to filter by peak height). Alternatively, the
<code>filter_peaks</code> function can be used to filter peaks after the
<code>peak_list</code> has already been created.</p>
</div>
<div class="section level4">
<h4 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h4>
<p>The <code>plot.peak_list</code> function allows you to visually
assess the peak integration accuracy. Below we compare the peaks fitted
for the first chromatogram (<code>idx = 1</code>) using the two
algorithms. Usually the egh function performs slightly better for
asymmetrical peaks. Even though the peaks are fully filled in when the
<code>raw</code> setting is selected, the results may not necessarily be
more accurate.</p>
<pre class="text"><code>&gt; par(mfrow=c(3,1))
&gt; plot(pks_gauss, idx = 1, lambda = 210)
&gt; legend("topright", "Gaussian", bty = "n")
&gt; 
&gt; plot(pks_egh, idx = 1, lambda = 210)
&gt; legend("topright", "Exponential-gaussian hybrid", bty = "n")
&gt; 
&gt; plot(pks_raw, idx = 1, lambda = 210)
&gt; legend("topright", "Raw", bty = "n")</code></pre>
<p><img src="chromatographR_files/figure-html/plot_peaks-1.png" width="600px"></p>
</div>
</div>
<div class="section level3">
<h3 id="peak-table-assembly">Peak table assembly<a class="anchor" aria-label="anchor" href="#peak-table-assembly"></a>
</h3>
<p>After obtaining a <code>peak_list</code>, the <code>get_peaks</code>
function performs complete-linkage hierarchical clustering to link peaks
across samples. It returns a <code>peak_table</code> object with samples
as rows and peaks as columns. The <code>peak_table</code> object also
has slots for holding metadata about peaks, samples, and the parameters
used in the analysis. If you have a lot of samples, this step can be
quite computationally expensive. Thus, it is suggested to filter the
peak_list provided to <code>get_peaktable</code> in order to remove
extraneous peaks (see <em>Peak finding and fitting</em> section above).
Peaks can also be filtered after <code>peak_table</code> assembly using
the <code>filter_peaktable</code> function. An important parameter here
is <code>hmax</code> which controls the stringency with which retention
times are matched across samples. A low value of <code>hmax</code> will
increase the odds of splitting a single peaks across multiple columns,
while a high value of <code>hmax</code> will increase the odds of
erroneously combining multiple peaks into the same column.</p>
<pre class="text"><code>&gt; # assembly peak table from peak_list saved in `pks_egh`
&gt; pk_tab &lt;- get_peaktable(pks_egh, response = "area", hmax = 0.2)
&gt; 
&gt; # print first six columns of peak table
&gt; head(pk_tab$tab[,1:6])</code></pre>
<pre><code>          V1        V2        V3       V4        V5        V6
119 5.496455 0.0000000 0.6572183 1.991365 15.199917 0.3114568
121 4.200859 1.0746359 0.6078366 0.000000  7.908693 0.0000000
122 8.261218 0.9544689 1.9510595 0.000000 25.772683 1.7912076
458 6.201729 2.2880101 1.9889770 5.308515 13.995000 0.2875473</code></pre>
</div>
<div class="section level3">
<h3 id="further-analysis-and-data-visualization">Further analysis and data-visualization<a class="anchor" aria-label="anchor" href="#further-analysis-and-data-visualization"></a>
</h3>
<div class="section level4">
<h4 id="attaching-metadata">Attaching metadata<a class="anchor" aria-label="anchor" href="#attaching-metadata"></a>
</h4>
<p>To begin analyzing your peak table, you will usually want to attach
sample metadata to the <code>peak_table</code> object. This can be
easily accomplished using the <code>attach_metadata</code> function.
This function takes a <code>metadata</code> argument which should be
supplied with a <code>data.frame</code> containing experimental
metadata, where one of the columns matches the names of your samples.
This column should be specified by supplying the column name as a string
to the <code>column</code> argument. This will attach the ordered
metadata in the <code>sample_meta</code> slot of your peak table. The
peak table can then be normalized (e.g. by dividing out the sample
weight) using the <code>normalize_data</code> function.</p>
<pre class="text"><code>&gt; # load example metadata
&gt; path &lt;- system.file("extdata", "Sa_metadata.csv", package = "chromatographR")
&gt; meta &lt;- read.csv(path)
&gt; # attach metadata
&gt; pk_tab &lt;- attach_metadata(peak_table = pk_tab, metadata = meta, column = "vial")
&gt; # normalize peak table by sample mass
&gt; pk_tab &lt;- normalize_data(peak_table = pk_tab, column = "mass")</code></pre>
</div>
<div class="section level4">
<h4 id="attaching-reference-spectra">Attaching reference spectra<a class="anchor" aria-label="anchor" href="#attaching-reference-spectra"></a>
</h4>
<p>Optionally, you can attach scaled reference spectra to the peak_table
using the <code>attach_ref_spectra</code> function. This can be helpful
for working with UV spectra programmatically (e.g. to sort peaks by
their chromophores). Reference spectra are defined either as the
spectrum with the highest intensity for each peak (when
<code>ref = "max.int"</code>) or as the spectrum with the highest
average correlation to the other spectra associated with the peak (when
<code>ref = "max.cor"</code>). Below, we show how these spectra can be
used to construct a correlation matrix to find peaks matching a
particular chromophore.</p>
<pre class="text"><code>&gt; pk_tab &lt;- attach_ref_spectra(pk_tab, ref = "max.int")
&gt; cor_matrix &lt;- cor(pk_tab$ref_spectra)
&gt; hx &lt;- names(which(cor_matrix[,"V20"] &gt; .99))
&gt; matplot(x = as.numeric(rownames(pk_tab$ref_spectra)),
+                    y = pk_tab$ref_spectra[, hx], type = 'l',
+                    ylab = "Abs (mAU)", 
+                    xlab = "Wavelength (nm)")</code></pre>
<p><img src="chromatographR_files/figure-html/unnamed-chunk-8-1.png" width="600px"></p>
</div>
<div class="section level4">
<h4 id="data-visualization">Data visualization<a class="anchor" aria-label="anchor" href="#data-visualization"></a>
</h4>
<div class="section level5">
<h5 id="mirror-plot">Mirror plot<a class="anchor" aria-label="anchor" href="#mirror-plot"></a>
</h5>
<p>The <code>mirror_plot</code> function provides a quick way to
visually compare results across treatment groups.</p>
<pre class="text"><code>&gt; mirror_plot(pk_tab, lambdas = c(210), var = "trt", legend_size = 2)</code></pre>
<p><img src="chromatographR_files/figure-html/mirror_plot-1.png" width="600px"></p>
</div>
<div class="section level5">
<h5 id="plotting-spectra">Plotting spectra<a class="anchor" aria-label="anchor" href="#plotting-spectra"></a>
</h5>
<p>The <code>plot_spectrum</code> function allows you to easily plot or
record the spectra associated with a particular peak in your peak table.
This is useful for interpreting your results and/or checking for errors
in your peak table. For example, you may want to check if the spectra
for a particular peak match across different samples, or you may want to
compare your spectrum with a known standard. The
<code>plot_spectrum</code> function can be used to plot only the
spectrum or only the chromatographic trace using the arguments
<code>plot_spectrum</code> and <code>plot_trace</code>. By default it
will plot the trace and spectrum from the chromatogram with the largest
peak in the peak table. Alternatively, you can choose the chromatogram
index and wavelength using the <code>idx</code> and <code>lambda</code>
arguments.</p>
<pre class="text"><code>&gt; par(mfrow = c(2,1))
&gt; peak &lt;- "V7"
&gt; plot_spectrum(peak, peak_table = pk_tab, chrom_list = warp, 
+               verbose = FALSE)</code></pre>
<p><img src="chromatographR_files/figure-html/plot_spectra-1.png" width="600px"></p>
<p>The <code>plot_spectrum</code> function can also be used to generate
interactive plots using <a href="https://plotly.com/r/" class="external-link">plotly</a>.</p>
<pre class="text"><code>&gt; plot_spectrum(peak, peak_table = pk_tab, chrom_list = warp, 
+               verbose = FALSE, engine = "plotly")</code></pre>
<p>The <code>plot_all_spectra</code> function can be used to visually
compare the spectra for a specified peak across all samples.</p>
<pre class="text"><code>&gt; peak &lt;- "V13"
&gt; plot_all_spectra(peak, peak_table = pk_tab, export = FALSE)</code></pre>
<p><img src="chromatographR_files/figure-html/plot_all_spectra-1.png" width="600px"></p>
</div>
<div class="section level5">
<h5 id="plot-peak-table-function">Plot peak table function<a class="anchor" aria-label="anchor" href="#plot-peak-table-function"></a>
</h5>
<p>The <code>plot.peak_table</code> function provides a simplified
interface to various options for plotting data from the
<code>peak_table</code>. For example, it can be used as a quick
interface to the <code>plot_spectrum</code> and
<code>plot_all_spectra</code> functions shown above. It can also be used
to quickly compare results across treatments by calling
<code>boxplot</code>.</p>
<pre class="text"><code>&gt; plot(pk_tab, loc = "V13", box_plot = TRUE, vars = "trt", verbose = FALSE)</code></pre>
<p><img src="chromatographR_files/figure-html/plot_peaktable-1.png" width="600px"><img src="chromatographR_files/figure-html/plot_peaktable-2.png" width="600px"><img src="chromatographR_files/figure-html/plot_peaktable-3.png" width="600px"></p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-bloemberg2010" class="csl-entry">
Bloemberg, Tom G., Jan Gerretzen, Hans J. P. Wouters, Jolein Gloerich,
Maurice van Dael, Hans J. C. T. Wessels, Lambert P. van den Heuvel, Paul
H. C. Eilers, Lutgarde M. C. Buydens, and Ron Wehrens. 2010.
<span>“Improved Parametric Time Warping for Proteomics.”</span>
<em>Chemometrics and Intelligent Laboratory Systems</em>,
<span>OMICS</span>, 104 (1): 65–74. <a href="https://doi.org/10.1016/j.chemolab.2010.04.008" class="external-link">https://doi.org/10.1016/j.chemolab.2010.04.008</a>.
</div>
<div id="ref-clifford2012" class="csl-entry">
Clifford, David, and Glenn Stone. 2012. <span>“Variable <span>Penalty
Dynamic Time Warping Code</span> for <span>Aligning Mass Spectrometry
Chromatograms</span> in <span>R</span>.”</span> <em>Journal of
Statistical Software</em> 47 (April): 1–17. <a href="https://doi.org/10.18637/jss.v047.i08" class="external-link">https://doi.org/10.18637/jss.v047.i08</a>.
</div>
<div id="ref-clifford2009" class="csl-entry">
Clifford, David, Glenn Stone, Ivan Montoliu, Serge Rezzi,
François-Pierre Martin, Philippe Guy, Stephen Bruce, and Sunil Kochhar.
2009. <span>“Alignment <span>Using Variable Penalty Dynamic Time
Warping</span>.”</span> <em>Analytical Chemistry</em> 81 (3): 1000–1007.
<a href="https://doi.org/10.1021/ac802041e" class="external-link">https://doi.org/10.1021/ac802041e</a>.
</div>
<div id="ref-eilers2004" class="csl-entry">
Eilers, Paul H. C. 2004. <span>“Parametric <span>Time
Warping</span>.”</span> <em>Analytical Chemistry</em> 76 (2): 404–11. <a href="https://doi.org/10.1021/ac034800e" class="external-link">https://doi.org/10.1021/ac034800e</a>.
</div>
<div id="ref-lan2001" class="csl-entry">
Lan, Kevin, and James W. Jorgenson. 2001. <span>“A Hybrid of Exponential
and Gaussian Functions as a Simple Model of Asymmetric Chromatographic
Peaks.”</span> <em>Journal of Chromatography A</em> 915 (1): 1–13. <a href="https://doi.org/10.1016/S0021-9673(01)00594-5" class="external-link">https://doi.org/10.1016/S0021-9673(01)00594-5</a>.
</div>
<div id="ref-tomasi2004" class="csl-entry">
Tomasi, Giorgio, Frans van den Berg, and Claus Andersson. 2004.
<span>“Correlation Optimized Warping and Dynamic Time Warping as
Preprocessing Methods for Chromatographic Data.”</span> <em>Journal of
Chemometrics</em> 18 (5): 231–41. <a href="https://doi.org/10.1002/cem.859" class="external-link">https://doi.org/10.1002/cem.859</a>.
</div>
<div id="ref-uesugi2013" class="csl-entry">
Uesugi, Akane, and André Kessler. 2013. <span>“Herbivore Exclusion
Drives the Evolution of Plant Competitiveness via Increased
Allelopathy.”</span> <em>New Phytologist</em> 198 (3): 916–24. <a href="https://doi.org/10.1111/nph.12172" class="external-link">https://doi.org/10.1111/nph.12172</a>.
</div>
<div id="ref-wehrens2015" class="csl-entry">
Wehrens, Ron, Elisabete Carvalho, and Paul D. Fraser. 2015.
<span>“Metabolite Profiling in <span>LC-DAD</span> Using Multivariate
Curve Resolution: The Alsace Package for <span>R</span>.”</span>
<em>Metabolomics</em> 11 (1): 143–54. <a href="https://doi.org/10.1007/s11306-014-0683-5" class="external-link">https://doi.org/10.1007/s11306-014-0683-5</a>.
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre class="text"><code>&gt; sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] chromatographR_0.7.2 knitr_1.48          

loaded via a namespace (and not attached):
 [1] sass_0.4.9            utf8_1.2.4            generics_0.1.3       
 [4] tidyr_1.3.1           bitops_1.0-9          xml2_1.3.6           
 [7] fastcluster_1.2.6     stringi_1.8.4         lattice_0.22-6       
[10] digest_0.6.37         magrittr_2.0.3        caTools_1.18.3       
[13] chromConverter_0.2.1  evaluate_1.0.1        grid_4.4.2           
[16] dynamicTreeCut_1.63-1 fastmap_1.2.0         cellranger_1.1.0     
[19] jsonlite_1.8.9        Matrix_1.7-1          Formula_1.2-5        
[22] purrr_1.0.2           fansi_1.0.6           scales_1.3.0         
[25] pbapply_1.7-2         textshaping_0.4.0     jquerylib_0.1.4      
[28] cli_3.6.3             rlang_1.1.4           munsell_0.5.1        
[31] ptw_1.9-16            cachem_1.1.0          yaml_2.3.10          
[34] tools_4.4.2           minpack.lm_1.2-4      dplyr_1.1.4          
[37] colorspace_2.1-1      RcppDE_0.1.7          reticulate_1.39.0    
[40] vctrs_0.6.5           R6_2.5.1              png_0.1-8            
[43] lifecycle_1.0.4       stringr_1.5.1         fs_1.6.5             
[46] htmlwidgets_1.6.4     ragg_1.3.3            pkgconfig_2.0.3      
[49] desc_1.4.3            pkgdown_2.1.1.9000    bslib_0.8.0          
[52] pillar_1.9.0          VPdtw_2.2.1           glue_1.8.0           
[55] Rcpp_1.0.13-1         systemfonts_1.1.0     highr_0.11           
[58] tidyselect_1.2.1      xfun_0.49             tibble_3.2.1         
[61] farver_2.1.2          htmltools_0.5.8.1     rmarkdown_2.28       
[64] compiler_4.4.2        readxl_1.4.3         </code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ethan Bass.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
